src/box/memtx_tree.cc

```diff
@@ -407,11 +407,71 @@ tree_iterator_prev_equal_base(struct iterator *iterator, struct tuple **ret)
 	return 0;
 }
 
+const char *
+to_mp_array(struct txn *txn, const char *key, uint32_t part_count)
+{
+	if (!txn || !key)
+		return key;
+
+	const char *end = key;
+	for (size_t i = 0; i < part_count; ++i) {
+		mp_next(&end);
+	}
+
+	size_t size = end - key + mp_sizeof_array(part_count);
+	char *new_key = (char *) region_alloc(&txn->region, size);
+
+	char *begin = new_key;
+
+	new_key = mp_encode_array(new_key, part_count);
+	memcpy((void *) new_key, key, end - key);
+
+	return begin;
+}
+
+template <bool USE_HINT>
+static int
+memtx_read_iterator_track_read(struct txn *txn,
+			       struct tree_iterator<USE_HINT> *itr,
+			       struct memtx_entry entry)
+{
+	if (!memtx_tx_manager_use_mvcc_engine)
+		return 0;
+	if (txn == NULL)
+		return 0;
+
+	// Point-like selects that found tuple don't need to be added in
+	// interval tree
+	if (itr->type == ITER_EQ && entry.key != NULL)
+		return 0;
+
+	memtx_entry key_entry;
+	key_entry.key = to_mp_array(txn, itr->key_data.key,
```

Comment:
Here i think may be a memory leak. to_mp_array allocates memory using region alloc, where it's frees using region truncate?